src/zenml/materializers/pandas_materializer.py

```diff
@@ -125,9 +138,20 @@ def save(self, df: Union[pd.DataFrame, pd.Series]) -> None:
 
             df = df.to_frame(name="series")
 
+        # Create a temporary file to store the data
         if self.pyarrow_exists:
-            with fileio.open(self.parquet_path, mode="wb") as f:
-                df.to_parquet(f, compression=COMPRESSION_TYPE)
+            with tempfile.NamedTemporaryFile(
+                mode="wb", suffix=".gzip", delete=False
+            ) as f:
+                df.to_parquet(f.name, compression=COMPRESSION_TYPE)
+                fileio.copy(f.name, self.parquet_path)
         else:
-            with fileio.open(self.csv_path, mode="wb") as f:
-                df.to_csv(f, index=True)
+            with tempfile.NamedTemporaryFile(
+                mode="wb", suffix=".csv", delete=False
+            ) as f:
+                df.to_csv(f.name, index=True)
+                fileio.copy(f.name, self.csv_path)
+
+        # Close and remove the temporary file
+        f.close()
```

Comment:
I think this isn't needed, the `with ...` context manager should already close the file automatically?